test/unit-test/dawn/Optimizer/Passes/TestPassComputeStageExtents.cpp

```diff
@@ -0,0 +1,213 @@
+//===--------------------------------------------------------------------------------*- C++ -*-===//
+//                          _
+//                         | |
+//                       __| | __ ___      ___ ___
+//                      / _` |/ _` \ \ /\ / / '_  |
+//                     | (_| | (_| |\ V  V /| | | |
+//                      \__,_|\__,_| \_/\_/ |_| |_| - Compiler Toolchain
+//
+//
+//  This file is distributed under the MIT License (MIT).
+//  See LICENSE.txt for details.
+//
+//===------------------------------------------------------------------------------------------===//
+
+#include "dawn/Compiler/DawnCompiler.h"
+#include "dawn/Compiler/Options.h"
+#include "dawn/SIR/SIR.h"
+#include "dawn/SIR/SIRSerializer.h"
+#include "test/unit-test/dawn/Optimizer/Passes/TestEnvironment.h"
+#include <fstream>
+#include <gtest/gtest.h>
+#include <streambuf>
+
+using namespace dawn;
+
+namespace {
+
+TEST(ComputeStageExtents, test_stencil_01) {
+  std::string filename = TestEnvironment::path_ + "/compute_extent_test_stencil_01.sir";
```

Comment:
should we denote the serialized stencil somewhere in the comments to make it, such that tests are better understood on their own?

Reply:
yes, exactly. They are generated from there. The idea is that everytime SIR changes we would need to regenerate them with gtclang. We would need a simple python script that generate all the SIR references used in testing of dawn from gtclang sources